shared/bsp/bsp_usb.cc

```diff
@@ -20,27 +20,130 @@
 
 #include "bsp_usb.h"
 
-static usb_callback_t rm_usb_rx_callback = NULL;
+#include "usbd_cdc_if.h"
 
-void usb_register_callback(const usb_callback_t callback) {
-  rm_usb_rx_callback = callback;
+#include "cmsis_os.h"
+
+static bsp::USB* usb = nullptr;
+
+namespace bsp {
+
+USB::USB() :
+    rx_size_(0), rx_pending_(0), rx_write_(nullptr), rx_read_(nullptr),
+    tx_size_(0), tx_pending_(0), tx_write_(nullptr), tx_read_(nullptr) {
+  if (usb)
+    delete usb;
```

Comment:
Still think this might be unsafe. IMO is better to do `RM_ASSERT_TRUE(usb == nullptr)`